### PR TITLE
Added support for Google buckets storage

### PR DESCRIPTION
**Pull request by bgrainger**
This replaces PR #172 (which hasn't seen active development recently).

- [x] Squash all commits in #172 and resolve conflicts with `master`
- [x] Addressing code review from #172
- [x] Update documentation

### CODE REVIEW

**Comment by bgrainger** [comment]
I'm currently running this code on Google App Engine, and it's working well. (URL available on request.)

File: docs/cloud/gcp.md [hunk 1]
```diff
@@ -1,9 +1,43 @@
...
+
+## Google Cloud SQL
+
+* TODO
+
+## Google AppEngine
+
+* TODO
+
+We're open source and accept contributions! [Fork us on GitHub](https://github.com/loic-sharma/BaGet).
```
Comment (loic-sharma) [inline docs/cloud/gcp.md]: Please move line 43 to somewhere after line 4

**Comment by loic-sharma** [comment]
The code looks great! I'm not very familiar with Google Cloud, is there an emulator I can run on Windows/MacOS that you'd recommend to test these changes?

File: docs/cloud/gcp.md (hunk 1 above)
Comment (loic-sharma) [reply docs/cloud/gcp.md]: Thanks for taking care of the docs by the way! 👍

**Comment by bgrainger** [comment]
There's no emulator (for Cloud Storage) officially supported by Google that I know of; there are only there four (for different products, and in beta, no less): https://cloud.google.com/sdk/gcloud/reference/beta/emulators/

The relevant issue appears to be https://github.com/googleapis/google-cloud-python/issues/4840.

I was able to test this as my employer is considering using BaGet and we have a GCP account.

File: src/BaGet.GCP/Configuration/GoogleCloudStorageOptions.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+using BaGet.Core.Configuration;
+
+namespace BaGet.GCP.Configuration
+{
+    public class GoogleCloudStorageOptions : StorageOptions
+    {
+        public string BucketName { get; set; }
```
Comment (loic-sharma) [inline src/BaGet.GCP/Configuration/GoogleCloudStorageOptions.cs]: Please mark this as `[Required]`. For example, see: https://github.com/loic-sharma/BaGet/blob/master/src/BaGet.Azure/Configuration/BlobStorageOptions.cs#L7

File: src/BaGet.GCP/Services/GoogleCloudStorageService.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+namespace BaGet.GCP.Services
+{
+    public class GoogleCloudStorageService : IStorageService
+    {
+        private readonly string _bucketName;
+
+        public GoogleCloudStorageService(IOptionsSnapshot<GoogleCloudStorageOptions> options)
+        {
+            _bucketName = options.Value.BucketName;
```
Comment (loic-sharma) [inline src/BaGet.GCP/Services/GoogleCloudStorageService.cs]: Please add `if (options == null) throw new ArgumentNullException(nameof(options));` before storing the `_bucketName`

Comment (bgrainger) [reply src/BaGet.GCP/Services/GoogleCloudStorageService.cs]: Will do.

BTW, was just copying this existing code; does it need to be fixed? https://github.com/loic-sharma/BaGet/blob/873b133c619e2bc048bbedc5a7fa8007c688eaf8/src/BaGet.AWS/S3StorageService.cs#L22

Comment (bgrainger) [reply src/BaGet.GCP/Services/GoogleCloudStorageService.cs]: Maybe never mind; that's `IOptions<>` vs `IOptionsSnapshot<>` and I don't know the difference (specifically in use and nullability).

Comment (loic-sharma) [reply src/BaGet.GCP/Services/GoogleCloudStorageService.cs]: Good catch. Yes, both should have null check and both should use `IOptionsSnapshot`. Feel free to fix it in this pull request if you'd like. If not, I can take care of it later

Comment (bgrainger) [reply src/BaGet.GCP/Services/GoogleCloudStorageService.cs]: I can push a new commit to add that check.

**Comment by bgrainger** [comment]
All code review was addressed and `null` checks added to `S3StorageService`; this PR is ready for final review.

**Comment by bgrainger** [comment]
@loic-sharma Just checking that you're not waiting on me for anything more before this can be merged?

**Comment by loic-sharma** [comment]
Looks great, thanks @bgrainger!

**Pull request merged**